samples/nrf9160/modem_shell/src/sock/sock.c

```diff
@@ -336,7 +336,6 @@ static int sock_getaddrinfo(
 				"getaddrinfo() failed, err %d errno %d",
 				err,
 				errno);
-			err = errno;
```

Comment:
The reason why this doesn't work is that I get error return value -11 and at the same time errno is set to 0.
@lemrey Can you comment if this is normal or a potential bug in libmodem?

Reply:
Could be a bug in nrf91_sockets.c. If the error code returned by `nrf_getaddrinfo()` doesn't match any of the error codes listed in `nrf_to_z_dns_error_code()`, `DNS_EAI_SYSTEM` will be returned by the default branch but `errno` is not set.

Reply:
yeah, it looks like `nrf91_socket.c` needs some updates.

Reply:
See if this helps: https://github.com/nrfconnect/sdk-nrf/pull/5675

Reply:
Still, `errno` shall only be set for `DNS_EAI_SYSTEM` - in case any other error code is returned by `getaddrinfo()` `errno` will be unspecified, I don't think it should be logged in such case. 

Plus, we're mixing `DNS_EAI_` error space with `errno` values (see -EINVAL above) which leaves ambiguity.

Reply:
Fixing the logging part.

True about mixing the error codes. While it doesn't really matter here as the error code is not really used anywhere, I will set `err = -EADDRNOTAVAIL` whenever getaddrinfo fails.